### PR TITLE
docs(ingress-nginx): correct nginx mentionings and replace NGINX Ingress usage

### PR DESCRIPTION
**Pull request by Langleu**
## Description

related to https://github.com/camunda/team-infrastructure-experience/issues/24.

[ingress-nginx Controller](https://github.com/kubernetes/ingress-nginx) and [NGINX Ingress Controller](https://docs.nginx.com/nginx-ingress-controller/) are often homogeneously used and abstracted by just `nginx`, `ingress controller`, `nginx ingress controller`. All of which results in confusion of which one to use as both have different target groups and histories.

The Camunda 8 Helm chart heavily focuses on `ingress-nginx`, which results in e.g. `NGINX Ingress Controller` not working out of the box since first it doesn't have `http2` enabled out of the box = no grpc and second annotations are missing to make it work. It can be done but we have nothing documented about it.

This PR aims to fix the confusion and just stick to `ingress-nginx`. Any other mentioning was adjusted to be specific about the used controller. Additionally switches the usage of `NGINX Ingress Controller` for `ingress-nginx` as in theory it could have never worked with the mentioned documentation.

Lastly, for OpenShift the paragraph about alternatives was extended and an example was referenced which could be enough to make it work with `Ingress NGINX Controller`, the reason for keeping is due to Red Hat officially endorsing this controller for OpenShift. For `ingress-nginx` I would require tests to confirm that it works as their docs don't mention anything and I could imagine some security fixes are required to make it work under OpenShift.

## When should this change go live?

- [ ] This change is not yet live and should not be merged until {ADD_DATE} (apply `hold` label or convert to draft PR)?
- [x] There is no urgency with this change.
- [ ] This change or page is part of a marketing blog, conference talk, or something else on a schedule.
- [ ] This functionality is already available but undocumented.
- [x] This is a bug fix ~or security concern.~

## PR Checklist

- [x] I have added changes to the relevant `/versioned_docs` directory, or they are not for an **already released version**.
- [x] I have added changes to the main `/docs` directory (aka `/next/`), or they are not for **future versions**.
- [x] My changes require an [Engineering review](https://github.com/camunda/camunda-docs/blob/main/howtos/documentation-guidelines.md#review-process), and I've assigned an engineering manager or tech lead as a reviewer, or my changes do not require an Engineering review.
- [ ] My changes require a [technical writer review](https://github.com/camunda/camunda-docs/blob/main/howtos/documentation-guidelines.md#review-process), and I've assigned @christinaausley as a reviewer, or my changes do not require a technical writer review.

### CODE REVIEW

**Comment by aabouzaid** [comment]
Thanks Lars for that fix :raised_hands: 

I just have one note, in the official end-user docs in the body (not the dev repo and not in the cli) like [How it works](https://kubernetes.github.io/ingress-nginx/how-it-works/) the name used to refer to the controller is `Ingress-Nginx Controller`.

**Comment by maxdanilov** [comment]
Also not sure about the name, but well they're so inconsistent about it (I've spotted 4 different spellings at least) I don't think one can be correct here:

From here: https://kubernetes.github.io/ingress-nginx/how-it-works/

- Screenshot 2024-01-03 at 16 41 47
- Screenshot 2024-01-03 at 16 41 55

From https://github.com/kubernetes/ingress-nginx

- Screenshot 2024-01-03 at 16 42 32
- Screenshot 2024-01-03 at 16 42 39
- Screenshot 2024-01-03 at 16 43 16

File: docs/self-managed/platform-deployment/helm-kubernetes/platforms/redhat-openshift.md [hunk 1]
```diff
@@ -269,7 +269,7 @@ To use these routes for the Zeebe Gateway, configure this through Ingress as wel
 
 ### Alternatives
 
-An alternative is to install the Ingress Controller of choice and use this instead; for example, [NGINX](https://www.redhat.com/en/blog/using-nginx-ingress-controller-red-hat-openshift).
+An alternative is to install the Ingress Controller of choice and use this instead. This may require additional adjustments on the Camunda 8 ingress since it heavily focuses on the [ingress-nginx Controller](https://github.com/kubernetes/ingress-nginx). Do not confuse the controller with the Red Hat endorsed [NGINX Ingress Controller](https://www.redhat.com/en/blog/using-nginx-ingress-controller-red-hat-openshift). The latter would require additional adjustments on the Camunda 8 ingress and `NGINX Ingress Controller` to make `grpc` and `http2` work, please have a look at this [example and the prerequisites](https://github.com/nginxinc/kubernetes-ingress/blob/main/examples/ingress-resources/grpc-services/README.md).
```
Comment (maxdanilov) [inline docs/self-managed/platform-deployment/helm-kubernetes/platforms/redhat-openshift.md:272]: I would rephrase this a bit for better clarity:

```suggestion
An alternative to using [routes](https://docs.openshift.com/container-platform/4.14/networking/routes/route-configuration.html) is to install and use one of the [Kubernetes Ingress Controllers](https://kubernetes.io/docs/concepts/services-networking/ingress-controllers/) instead. The de-facto standard in the industry is the [ingress-nginx Controller](https://github.com/kubernetes/ingress-nginx). 

Do not confuse the [ingress-nginx controller](https://github.com/kubernetes/ingress-nginx) with the [NGINX Ingress Controller](https://www.redhat.com/en/blog/using-nginx-ingress-controller-red-hat-openshift) that is endorsed by Red Hat for usage with OpenShift.

If you use [NGINX Ingress Controller](https://www.redhat.com/en/blog/using-nginx-ingress-controller-red-hat-openshift), you would require additional adjustments done to the Camunda 8 ingress objects and the NGINX Ingress Controlle itself, to make `gRPC` and `HTTP/2` connections work. In that case, please refer to the [example and the prerequisites](https://github.com/nginxinc/kubernetes-ingress/blob/main/examples/ingress-resources/grpc-services/README.md).
```

Comment (maxdanilov) [reply docs/self-managed/platform-deployment/helm-kubernetes/platforms/redhat-openshift.md:272]: I'd also recommend using something like a [warning admonition](https://docusaurus.io/docs/markdown-features/admonitions) for the part that brings attention to the difference between the nginx-ingress and ingress-nginx (as otherwise a reader may not pay attention to it - the naming difference between the two is minimal)

**Review by maxdanilov — CHANGES_REQUESTED** [review]
I've left some minor comments for the latest version (but it also applies to the versioned ones)

File: docs/self-managed/platform-deployment/helm-kubernetes/guides/ingress-setup.md [hunk 2]
```diff
@@ -191,21 +191,31 @@ Ingress resources require the cluster to have an [Ingress Controller](https://ku
...
+      maxUnavailable: 1
   service:
     type: NodePort
+
+  publishService:
+    enabled: false
 ```
 
-To install this Ingress Controller to your local cluster, execute the following command:
+To install this `ingress-nginx Controller` to your local cluster, execute the following command:
```
[the remark this answers is not in the document]
Comment (Langleu) [reply docs/self-managed/platform-deployment/helm-kubernetes/guides/ingress-setup.md:210]: Good point 👍

[the remark this answers is not in the document]
Comment (Langleu) [reply docs/self-managed/platform-deployment/helm-kubernetes/guides/ingress-setup.md:210]: yeah, wish there was an easier way to maintain the versioned docs ... it will only get more complicated for each version, especially infrastructure which is usually generally applicable independent of C8.

File: docs/self-managed/platform-deployment/helm-kubernetes/platforms/redhat-openshift.md (hunk 1 above)
Comment (Langleu) [reply docs/self-managed/platform-deployment/helm-kubernetes/platforms/redhat-openshift.md:272]: image

**Review by maxdanilov — APPROVED** [review]
Looks great, thank you!

File: docs/self-managed/platform-deployment/helm-kubernetes/platforms/redhat-openshift.md [hunk 3]
```diff
@@ -269,7 +269,15 @@ To use these routes for the Zeebe Gateway, configure this through Ingress as wel
...
+An alternative to using [routes](https://docs.openshift.com/container-platform/4.14/networking/routes/route-configuration.html) is to install and use one of the Kubernetes Ingress Controllers instead, for example, the [ingress-nginx Controller](https://github.com/kubernetes/ingress-nginx).
+
+:::warning
+
+Do not confuse the [ingress-nginx Controller](https://github.com/kubernetes/ingress-nginx) with the [NGINX Ingress Controller](https://www.redhat.com/en/blog/using-nginx-ingress-controller-red-hat-openshift) that is endorsed by Red Hat for usage with OpenShift. Despite very similar names, they are two different products.
+
+If you should decide to use the Rad Hat endorsed [NGINX Ingress Controller](https://www.redhat.com/en/blog/using-nginx-ingress-controller-red-hat-openshift), you would require additional adjustments done to the Camunda 8 ingress objects and the NGINX Ingress Controlle itself, to make `gRPC` and `HTTP/2` connections work. In that case, please refer to the [example and the prerequisites](https://github.com/nginxinc/kubernetes-ingress/blob/main/examples/ingress-resources/grpc-services/README.md).
```
Comment (akeller) [inline docs/self-managed/platform-deployment/helm-kubernetes/platforms/redhat-openshift.md:278]: Looks like there is a typo here, but Rad Hat sounds like a sweet company name 😆

**Comment by akeller** [comment]
Hey y'all, unless you are making code snippet changes only, please include @christinaausley as a reviewer for grammar, style, consistency, and typos. This will also help her identify things that should be added to the glossary or style guide incrementally.

**Comment by Langleu** [comment]
thanks @akeller! Probably makes sense to change the PR checklist / documentation guidelines as so far it sounds optional.
